integration/common.go

```diff
@@ -0,0 +1,112 @@
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/pelletier/go-toml"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	cri "k8s.io/cri-api/pkg/apis"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// RegistryList holds public image registries
+type RegistryList struct {
+	DockerLibraryRegistry    string
+	K8sCriContainerdRegistry string
+	K8sGcrRegistry           string
+}
+
+var (
+	imageService             cri.ImageManagerService
+	imageConfigs             map[int]string
+	dockerLibraryRegistry    string
+	k8sCriContainerdRegistry string
+	k8sGcrRegistry           string
+	pauseImage               string // This is the same with default sandbox image
+)
+
+func initReg(repoList string) {
+	registry := RegistryList{
+		DockerLibraryRegistry:    "docker.io/library",
+		K8sCriContainerdRegistry: "gcr.io/k8s-cri-containerd",
+		K8sGcrRegistry:           "k8s.gcr.io",
+	}
+
+	if repoList != "" {
+		fileContent, err := ioutil.ReadFile(repoList)
+		if err != nil {
+			panic(fmt.Errorf("Error reading '%v' file contents: %v", repoList, err))
+		}
+
+		err = toml.Unmarshal(fileContent, &registry)
+		if err != nil {
+			panic(fmt.Errorf("Error unmarshalling '%v' TOML file: %v", repoList, err))
+		}
+	}
```

Comment:
can we log the what we're going to be actually testing on before we return from initReg?

Reply:
Done